duffy/legacy/operations.py

```diff
@@ -0,0 +1,17 @@
+from fastapi import HTTPException
+from starlette.status import HTTP_403_FORBIDDEN
+
+from ..configuration import config
+from .api_models import Credentials
+
+
+def obtain_destination():
+    return config["metaclient"]["dest"].rstrip("/")
+
+
+def obtain_credentials(key: str = None):
+    userlist = config["metaclient"]["usermap"]
+    if key not in userlist:
+        raise HTTPException(HTTP_403_FORBIDDEN)
+    username, password = userlist[key], key
+    return Credentials(username=username, password=password)
```

Comment:
This won't work because the `key` won't be passed in properly (and default to `None).

Also, FastAPI dependencies usually are nouns, not verbs – e.g. we have `req_tenant` in the v1 API which means "the tenant on whose behalf this request is serviced" – so let's go along here and call it e.g. `req_credentials`. And `userlist` should be e.g. `usermap` because it isn't a list.

So, this should become something like (caution: untested):

```
from fastapi import Depends, HTTPException, Security
from fastapi.security import APIKeyQuery
from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_403_FORBIDDEN

from ..configuration import config
from .api_models import Credentials


def _req_credentials_factory(optional: bool = False):
    security = APIKeyQuery(name="key", auto_error=not optional)

    def _req_credentials(key: str = Security(security)):
        if not key:
            if not optional:
                raise HTTPException(HTTP_401_UNAUTHORIZED)
            else:
                return None

        usermap = config["metaclient"]["isermap"]
        if key not in usermap:
            raise HTTPException(HTTP_403_FORBIDDEN)
        username, password = usermap[key], key
        return Credentials(username=username, password=password)

    return _req_credentials

req_credentials = _req_credentials_factory()
req_credentials_optional = _req_credentials_factory(optional=True)
```